documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -11,203 +11,821 @@ Proposal API for backend-agnostic, refactored Input via keyboards, mice, and con
+    Number2,
+    Number3,
+    Number4,
+    Number5,
+    Number6,
+    Number7,
+    Number8,
+    Number9,
+    Semicolon /* ; */,
+    Equal /* = */,
+    A,
+    B,
+    C,
+    D,
+    E,
+    F,
+    G,
+    H,
+    I,
+    J,
+    K,
+    L,
+    M,
+    N,
+    O,
+    P,
+    Q,
+    R,
+    S,
+    T,
+    U,
+    V,
+    W,
+    X,
+    Y,
+    Z,
+    LeftBracket /* [ */,
+    BackSlash /* \ */,
+    RightBracket /* ] */,
+    GraveAccent /* ` */,
+    Escape,
+    Enter,
+    Tab,
+    Backspace,
+    Insert,
+    Delete,
+    Right,
+    Left,
+    Down,
+    Up,
+    PageUp,
+    PageDown,
+    Home,
+    End,
+    CapsLock,
+    ScrollLock,
+    NumLock,
+    PrintScreen,
+    Pause,
+    F1,
+    F2,
+    F3,
+    F4,
+    F5,
+    F6,
+    F7,
+    F8,
+    F9,
+    F10,
+    F11,
+    F12,
+    F13,
+    F14,
+    F15,
+    F16,
+    F17,
+    F18,
+    F19,
+    F20,
+    F21,
+    F22,
+    F23,
+    F24,
+    F25,
+    Keypad0,
+    Keypad1,
+    Keypad2,
+    Keypad3,
+    Keypad4,
+    Keypad5,
+    Keypad6,
+    Keypad7,
+    Keypad8,
+    Keypad9,
+    KeypadDecimal,
+    KeypadDivide,
+    KeypadMultiply,
+    KeypadSubtract,
+    KeypadAdd,
+    KeypadEnter,
+    KeypadEqual,
+    ShiftLeft,
+    ControlLeft,
+    AltLeft,
+    SuperLeft,
+    ShiftRight,
+    ControlRight,
+    AltRight,
+    SuperRight,
+    Menu
+}
+```
+
+The `KeyName` enum is exactly the same as the `Key` enum in 2.X. The integral values of each enumerant, not included here, must match the en-US scancode for that key. A backend must match a scancode to a `KeyName` as if it were an en-US scancode, as this is the keyboard layout from which these key names were derived.
+
+The Silk.NET team wishes to reserve the right to remove any key names which do not have a matching en-US scancode. This is because the above enum is just copied and pasted from 2.X, and has not been cross-referenced with the keyboard layout at this time.
+
+# Gamepad Input
+
+```cs
+public interface IGamepad : IInputDevice
+{
+    ref readonly GamepadState State { get; }
+    IReadOnlyList<IMotor> VibrationMotors { get; }
+}
+```
+
+`State` is the device state as defined earlier.
+
+`VibrationMotors` enumerates the vibration motors on the device. The values within `IMotor` do not change according to end user input, as such they are not encapsulated in the `State`, following the same principles as `ICursorConfiguration`.
+
+`IMotor` is defined as follows:
+```cs
+public interface IMotor
+{
+    float Speed { get; set; }
+}
+```
+
+This is exactly as in 2.X.
+
+**INFORMATIVE TEXT:** It is possible that, should integration with DualSense controllers be implemented, that more properties are added to `IMotor` than just `Speed`.
+
+`GamepadState` is defined as follows:
+```cs
+public readonly record struct GamepadState
+(
+    JoystickButtonState Buttons,
+    DualReadOnlyList<Vector2> Thumbsticks,
+    DualReadOnlyList<float> Triggers,
+    IReadOnlyList<IMotor> VibrationMotors
```

Comment:
`GamepadState` exposes the same `IMotor` as is already on `IGamepad`, but if the same `IMotor` instances are used then they could change on the next `Update()` (And also they're settable!)

I'd recommend removing these from `GamepadState` altogether (and JoystickState probably has something similar)